website/content/docs/enterprise/replication/check-merkle-tree-corruption.mdx

```diff
@@ -0,0 +1,309 @@
+
+## Merkle difference results in no delta 
+
+A merkle-diff operation resulting in no delta indicates conflicting Merkle tree pages. Despite the two clusters holding exactly the same data in both trees, their root hashes do not match.
+
+The following example log shows entries from a performance replication secondary indicating the issue resulting from a corrupted tree, and no writes on the primary since the previous merkle-sync operation.
+
+<CodeBlockConfig hideClipboard>
+
+```plaintext
+vault [INFO]  .perf-sec.core0.core: non-matching guard, exiting
+vault [TRACE] .perf-sec.core0.core: finished client WAL streaming
+vault [INFO]  .perf-sec.core0.replication: no matching WALs available
+vault [DEBUG] .perf-sec.core0.replication: starting merkle diff
+vault [TRACE] .perf-sec.core0.core: wal context done
+vault [TRACE] .perf-sec.core0.core: checking conflicting pages
+vault [TRACE] .perf-pri.core0.core: serving conflicting pages
+vault [DEBUG] .perf-pri.core0.replication.index.perf: creating merkle state snapshot: generation=4
+vault [DEBUG] .perf-pri.core0.replication.index.perf: removing state snapshot from cache: generation=4
+vault [INFO]  .perf-sec.core0.replication: requesting WAL stream: guard=8acf94ac
+vault [TRACE] .perf-sec.core0.core: starting client WAL streaming
+vault [TRACE] .perf-sec.core0.core: receiving WALs
+vault [TRACE] .perf-pri.core0.core: starting serving WALs: clientID=e16930a6-7d24-6924-41fe-aa8beb90b1b2
+vault [TRACE] .perf-pri.core0.core: streaming from log shipper done: clientID=e16930a6-7d24-6924-41fe-aa8beb90b1b2
+vault [TRACE] .perf-pri.core0.core: internal wal stream stop channel fired: clientID=e16930a6-7d24-6924-41fe-aa8beb90b1b2
+vault [TRACE] .perf-pri.core0.core: stopping serving WALs: clientID=e16930a6-7d24-6924-41fe-aa8beb90b1b2
+vault [INFO]  .perf-sec.core0.core: non-matching guard, exiting
+vault [TRACE] .perf-sec.core0.core: finished client WAL streaming
+vault [INFO]  .perf-sec.core0.replication: no matching WALs available
+vault [TRACE] .perf-sec.core0.core: wal context done
+vault [DEBUG] .perf-sec.core0.replication: starting merkle diff
+vault [TRACE] .perf-sec.core0.core: checking conflicting pages
+vault [TRACE] .perf-pri.core0.core: serving conflicting pages
+```
+
+</CodeBlockConfig>
+
+In the example log output, the performance secondary cluster's finite state machine (FSM) is entering merkle-diff mode, in which it tries to fetch conflicting pages from the primary cluster. The diff result is empty, indicated by an immediate switch to the stream-wals mode and **skipping the merkle-sync operation**. 
+
+Further into the log, the performance secondary cluster immediately goes into the merkle-diff mode again trying to reconcile the discrepancies of its Merkle tree with the primary cluster. This loop goes on without resolution due to the Merkle tree corruption.
+Non resolving merkle-sync
```

Comment:
Did you mean this to be a sub-section???

```suggestion

### Non resolving merkle-sync

```


Right now, it looks like an incomplete sentence. The [original run book](https://hashicorp.atlassian.net/wiki/spaces/VAULT/pages/2814280234/Merkle-Check+Runbook#Non-resolving-merkle-sync) has it as a header.

![image](https://github.com/hashicorp/vault/assets/7660718/8dba33e2-0b28-4160-ac07-667f1a082512)